### PR TITLE
Dedicated Rehydration Bucket

### PR DESCRIPTION
**Pull request by jfromme**
Adds a new S3 bucket to act as the destination for all rehydrations.

Restricts the Fargate rehydration task to read-only access to the source, published bucket.

Also some refactoring:

- Pass AWS service clients to `New*` functions instead of `aws.Config` so that clients can be shared
- Remove REGION and ENV from the model used to create RunTask inputs. These env vars are already present in the Fargate task definition, and don't need to be passed in each RunTask.

### CODE REVIEW

File: terraform/s3.tf [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["GET", "HEAD"]
+    allowed_origins = ["*"]
+    max_age_seconds = 3000
+  }
+}
+
+resource "aws_s3_bucket_acl" "rehydration_s3_acl" {
```
[the remark this answers is not in the document]
Comment (muftring) [reply terraform/s3.tf:31]: Here's a link to an AWS document regarding ACLs.

https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html?icmpid=docs_amazons3_console

That said, does using the "public-read" ACL make the access definition easier to define? When I created the Publishing 50 buckets, Terraform complained about the presence of an ACL.

File: terraform/s3.tf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  bucket     = aws_s3_bucket.rehydration_s3_bucket.id
+  acl        = "public-read"
+  depends_on = [aws_s3_bucket_ownership_controls.rehydration_s3_bucket_ownership_controls]
+}
+
+resource "aws_s3_bucket_ownership_controls" "rehydration_s3_bucket_ownership_controls" {
+  bucket = aws_s3_bucket.rehydration_s3_bucket.id
+  rule {
+    object_ownership = "ObjectWriter"
```
Comment (muftring) [inline terraform/s3.tf:40]: What's the motivation for using Object Writer here, rather than enforcing Bucket Owner?

**Review by muftring — APPROVED** [review]
approved, with a few comments/questions

File: terraform/s3.tf (hunk 2 above)
Comment (jfromme) [reply terraform/s3.tf]: Thanks for pointing this and the ACL thing out @muftring. Both were pure copy and paste, so no real motivation. I guess they are linked in that you can now only have ACLs if you don't enforce Bucket Owner. 

It looks like the bucket policy and the block public access parts already make this a public bucket, so I'm just going to delete the ACL and ownership controls stuff.

**Pull request merged**